website_sale_order_type/models/sale_order.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _cart_update(
```

Comment:
I'd say it would be better to use this prepare https://github.com/odoo/odoo/blob/12.0/addons/website_sale/models/website.py#L193-L219

Otherwise, your triggering the onchange on every cart update, which doesn't seem very optimum and could change some prefrences of the checkout like the shipping address...

Reply:
Ok, thank you. I hadn't thought of the repercussions of using this onchange method. The only change I want to make is to add the SO type, and I didn't consider that all the other fields would be updated as well.

I pushed an update following your suggestion. From what I have tested, the `_prepare_sale_order_values` method is not updating the values on the sale order as with the `_cart_update` method. What else would I need to do?

Reply:
Mmm, but the tests go fine, aren't they?